pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -314,6 +316,41 @@ func (p *staticPolicy) updateCPUsToReuse(pod *v1.Pod, container *v1.Container, c
 	p.cpusToReuse[string(pod.UID)] = p.cpusToReuse[string(pod.UID)].Difference(cset)
 }
 
+func (p *staticPolicy) ClearCPUsToReuse(pod *v1.Pod) {
+	if _, ok := p.cpusToReuse[string(pod.UID)]; ok {
+		delete(p.cpusToReuse, string(pod.UID))
+	}
+}
+
+func (p *staticPolicy) inUseCPUsForNonRestartableInitContainer(s state.State, pod *v1.Pod, container *v1.Container) cpuset.CPUSet {
+	var behindContainer = false
+	inUseCPUs := cpuset.New()
+
+	for _, initContainer := range pod.Spec.InitContainers {
+		if container.Name == initContainer.Name {
+			if podutil.IsRestartableInitContainer(&initContainer) {
+				// If the container is a restartable init container, we should not
+				// reuse its cpuset, as a restartable init container can run with
+				// regular containers.
+				break
+			} else {
+				behindContainer = true
+				continue
+			}
+		}
+		if behindContainer == true && podutil.IsRestartableInitContainer(&initContainer){
+			cset, _ := s.GetCPUSet(string(pod.UID), initContainer.Name)
+			inUseCPUs = inUseCPUs.Union(cset)
+		}
+	}
+	for _, c := range pod.Spec.Containers {
+		cset, _ := s.GetCPUSet(string(pod.UID), c.Name)
+		inUseCPUs = inUseCPUs.Union(cset)
+	}
+	
+	return inUseCPUs
+}
```

Comment:
here we want to find the cpus which were assigned to init container and are not transferred to app containers?
IOW, the intent is to find the CPUs leaked during the handoff between init and app containers?